Infura IPFS SSL certificate verification fails

Hello!

I just updated to `ipfshttpclient` from `py-ipfs-api` and now all Infura IPFS requests are failing.

Infura IPFS multiaddr:
`self.base_uri = "/dns/ipfs.infura.io/tcp/5001/https"` 
(I've also tried with `/dns4/`)

Connecting to Infura IPFS
```python
self.client = ipfshttpclient.connect(self.base_uri)
```

Stack trace: 
```sh
ethpm/backends/ipfs.py:60: in __init__
    self.client = ipfshttpclient.connect(self.base_uri)
venv/lib/python3.6/site-packages/ipfshttpclient/client/__init__.py:102: in connect
    assert_version(client.version()['Version'])
venv/lib/python3.6/site-packages/ipfshttpclient/client/base.py:15: in wrapper
    result = func(*args, **kwargs)
venv/lib/python3.6/site-packages/ipfshttpclient/client/miscellaneous.py:204: in version
    return self._client.request('/version', decoder='json', **kwargs)
venv/lib/python3.6/site-packages/ipfshttpclient/http.py:44: in wrapper
    return func(self, *args, **merged)
venv/lib/python3.6/site-packages/ipfshttpclient/http.py:364: in request
    files, headers, data, timeout=timeout)
venv/lib/python3.6/site-packages/ipfshttpclient/http.py:282: in _request
    timeout=timeout)
venv/lib/python3.6/site-packages/ipfshttpclient/http.py:249: in _do_request
    six.raise_from(exceptions.ConnectionError(error), error)
<string>:3: in raise_from
    ???
E   ipfshttpclient.exceptions.ConnectionError: SSLError: HTTPSConnectionPool(host='ipfs.infura.io', port=5001): Max retries exceeded with url: /api/v0/version?stream-channels=true (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))
```

I'm not sure what's the cause for the SSL certificate verification failure. 
There doesn't seem to be any problem with [infura's ssl certificates](https://www.sslshopper.com/ssl-checker.html#hostname=ipfs.infura.io). 
It appears that the multiaddr is being correctly parsed, since the request is being made to https://ipfs.infura.io:5001/api/v0/version. 

I've tried adding Infura's CA to `certifi`'s `cacert.pem` without any luck. 
I've also tried hard-coding `requests.request`'s `verify` parameter to `False` just to see if it works, but no luck.

Any thoughts or ideas on get these requests working again would be great, cheers!

--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@
 
 ![Python IPFS HTTP Client Library](https://ipfs.io/ipfs/QmQJ68PFMDdAsgCZvA1UVzzn18asVcf7HVvCDgpjiSCAse)
 
-Check out [the HTTP Client reference](https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/) for the full command reference.
+Check out [the HTTP Client reference](https://ipfs.io/ipns/12D3KooWEqnTdgqHnkkwarSrJjeMP2ZJiADWLYADaNvUb6SQNyPF/docs/) for the full command reference.
 
 **Important:** The `ipfsapi` PIP package and Python module have both been renamed to `ipfshttpclient`!
 See the [relevant section of the README](#important-changes-from-ipfsapi-04x) for details.
@@ -142,7 +142,7 @@ This module also contains some helper functions for adding strings and JSON to I
 
 Documentation (currently mostly API documentation unfortunately) is available on IPFS:
 
-https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/
+https://ipfs.io/ipns/12D3KooWEqnTdgqHnkkwarSrJjeMP2ZJiADWLYADaNvUb6SQNyPF/docs/
 
 The `ipfs` [command-line Client documentation](https://ipfs.io/docs/commands/) may also be useful in some cases.
 

--- a/ipfshttpclient/http.py
+++ b/ipfshttpclient/http.py
@@ -10,6 +10,7 @@ import abc
 import functools
 import tarfile
 from six.moves import http_client
+import os
 import socket
 try:  #PY3
 	import urllib.parse
@@ -23,7 +24,12 @@ import six
 
 from . import encoding
 from . import exceptions
-from . import requests_wrapper as requests
+PATCH_REQUESTS = (os.environ.get("PY_IPFS_HTTP_CLIENT_PATCH_REQUESTS", "yes").lower()
+                  not in ("false", "no"))
+if PATCH_REQUESTS:
+	from . import requests_wrapper as requests
+else:  # pragma: no cover (always enabled in production)
+	import requests
 
 
 def pass_defaults(func):
@@ -210,9 +216,9 @@ class HTTPClient(object):
 			query    = "",
 			fragment = ""
 		).geturl()
-		self._kwargs = {
-			"family": family
-		}
+		self._kwargs = {}
+		if PATCH_REQUESTS:  # pragma: no branch (always enabled in production)
+			self._kwargs["family"] = family
 		
 		self.defaults = defaults
 		self._session = None

--- a/ipfshttpclient/requests_wrapper.py
+++ b/ipfshttpclient/requests_wrapper.py
@@ -97,8 +97,9 @@ class ConnectionOverrideMixin:
 			extra_kw['socket_options'] = self.socket_options
 
 		try:
+			dns_host = getattr(self, "_dns_host", self.host)
 			conn = create_connection(
-				(self._dns_host, self.port), self.timeout, **extra_kw)
+				(dns_host, self.port), self.timeout, **extra_kw)
 		except socket.timeout:
 			raise urllib3.exceptions.ConnectTimeoutError(
 				self, "Connection to %s timed out. (connect timeout=%s)" %
@@ -128,7 +129,7 @@ class HTTPConnectionPool(urllib3.HTTPConnectionPool):
 	ConnectionCls = HTTPConnection
 
 
-class HTTPSConnectionPool(urllib3.HTTPConnectionPool):
+class HTTPSConnectionPool(urllib3.HTTPSConnectionPool):
 	ConnectionCls = HTTPSConnection
 
 
@@ -158,11 +159,14 @@ class PoolManager(urllib3.PoolManager):
 		if request_context is None:
 			request_context = self.connection_pool_kw.copy()
 		
+		for key in ('host', 'port'):
+			request_context.pop(key, None)
+		
 		if scheme == "http" or scheme.startswith("http+"):
 			for kw in urllib3.poolmanager.SSL_KEYWORDS:
 				request_context.pop(kw, None)
 		
-		return pool_cls(**request_context)
+		return pool_cls(host, port, **request_context)
 
 	def connection_from_pool_key(self, pool_key, request_context=None):
 		# Copied from `urllib3` so that we continue to ensure that this will

--- a/ipfshttpclient/version.py
+++ b/ipfshttpclient/version.py
@@ -8,4 +8,4 @@
 # `0.4.1` and so on. When IPFS `0.5.0` is released, the first client version
 # to support it will also be released as `0.5.0`.
 
-__version__ = "0.4.11"
+__version__ = "0.4.12"

--- a/renaming_status.txt
+++ b/renaming_status.txt
@@ -23,7 +23,7 @@ FOR README.md:
     https://travis-ci.org/ipfs/py-ipfs-api.svg?branch=master -> https://travis-ci.org/ipfs/py-ipfs-http-client.svg?branch=master        line 7
     https://travis-ci.org/ipfs/py-ipfs-api -> https://travis-ci.org/ipfs/py-ipfs-http-client                                            line 7
     https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
-                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
+                    https://ipfs.io/ipns/12D3KooWEqnTdgqHnkkwarSrJjeMP2ZJiADWLYADaNvUb6SQNyPF/docs/                 line 11,118
     The PIP package and Python module: ipfshttpclient
     The Github repo: py-ipfs-api -> py-ipfs-http-client
     PROJECT RENAME:        Python IPFS API -> Python IPFS HTTP Client
@@ -50,7 +50,7 @@ FOR pyproject.toml:
 * Changes:  module = "ipfsapi" -> module = "ipfshttpclient"             Line 6
             py-ipfs-api -> py-ipfs-http-client
             https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
-                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/
+                    https://ipfs.io/ipns/12D3KooWEqnTdgqHnkkwarSrJjeMP2ZJiADWLYADaNvUb6SQNyPF/docs/
 * Doubtful: Line 3
 
 

